common/src/API.php

```diff
@@ -63,6 +63,21 @@ private function buildURL($address, $parameters) {
         }
         return $address . '&' . $query_string;
     }
+
+    public function hasConsent(array $data): bool {
+        $permission_url = $this->buildURL(
+            sprintf(
+                'https://%s/api/order_permissions?order_number=%u&id=%u',
+                $this->api_domain,
+                $data['order'],
+                $this->shop_id
+            ),
+            ['id' => $this->shop_id, 'code' => $this->api_key]
+        );
+        $permission_response = Requests::post($permission_url, []);
```

Comment:
Did you even test this? 
The endpoint is for a `GET` request, and you are doing a `POST` request here, this probably does not work.
https://github.com/webwinkelkeur/dash/pull/2059/files 

You should test your code before submitting a PR.